### DISCUSSION TITLE
Question about MenuItem Template

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#10195] [Q&A]
MenuItem as a `HeaderedSelectingItemsControl` inherits from `SelectingItemsControl`, but not `HeaderedItemsControl`, which means the header is not properly handled in https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ItemsControl.cs#L343-L397 (I also checked all overrides, no luck). 
I think this is the root cause of MenuItem header issue #9696. Fix #10019 takes ItemTemplate as HeaderTemplate, which is not the way how `HeaderedItemsControl` handles it. So, is this expected or a bug? Should `HeaderedSelectingItemsControl` define the same `HeaderTemplate` as `HeaderedItemsControl` did?

**Comment by maxkatz6** [comment]
It's a bit confusing with namings. 
Speaking about `HeaderedSelectingItemsControl` isolately, I agree that it should follow the same behavior as `HeaderedItemsControl`.

But MenuItem has a header property but doesn't have a content. What would ItemTemplate be used there for?
MenuItem essentially is expected to behave like a content control.

**Reply by rabbitism** [reply]
@maxkatz6 MenuItem should behave like TreeItem
